ir/instr.cpp

```diff
@@ -1443,13 +1537,24 @@ StateValue FpTernaryOp::toSMT(State &s) const {
     };
     break;
   }
-
-  auto scalar = [&](const StateValue &a, const StateValue &b,
-                    const StateValue &c, const Type &ty) {
+  
+  scalar = [&](const StateValue &a, const StateValue &b,
```

Comment:
these can stay unchanged. no need to declare it upfront

Reply:
Leaving them as `auto` does not work as I later need to assign a different lambda to the `scalar` variable. Each lambda expression has a unique type. Assignment operators are actually explicitly deleted for lambdas with captures.

I've now moved the declarations to the same statement as the initial defintions.